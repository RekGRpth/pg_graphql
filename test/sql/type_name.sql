begin;
    create table account(id int primary key);
    create type post_status as enum ('published', 'unpublished');

    select
        (rec).type_kind,
        (rec).meta_kind,
        (rec).is_builtin,
        (rec).entity,
        (rec).graphql_type_id,
        graphql.type_name(rec) type_name
    from
        graphql._type rec
    order by
        (rec).entity,
        (rec).type_kind,
        graphql.type_name(rec);

rollback;
