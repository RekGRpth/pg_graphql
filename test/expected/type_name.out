begin;
    create table account(id int primary key);
    create type post_status as enum ('published', 'unpublished');
    select
        (rec).type_kind,
        (rec).meta_kind,
        (rec).is_builtin,
        (rec).entity,
        (rec).graphql_type_id,
        graphql.type_name(rec) type_name
    from
        graphql._type rec;
  type_kind   |      meta_kind      | is_builtin | entity  | graphql_type_id |       type_name       
--------------+---------------------+------------+---------+-----------------+-----------------------
 SCALAR       | ID                  | t          |         |                 | ID
 SCALAR       | Int                 | t          |         |                 | Int
 SCALAR       | Float               | t          |         |                 | Float
 SCALAR       | String              | t          |         |                 | String
 SCALAR       | Boolean             | t          |         |                 | Boolean
 SCALAR       | DateTime            | t          |         |                 | DateTime
 SCALAR       | BigInt              | t          |         |                 | BigInt
 SCALAR       | UUID                | t          |         |                 | UUID
 SCALAR       | JSON                | t          |         |                 | JSON
 SCALAR       | Cursor              | t          |         |                 | Cursor
 OBJECT       | Query               | t          |         |                 | Query
 OBJECT       | Mutation            | t          |         |                 | Mutation
 OBJECT       | PageInfo            | t          |         |                 | PageInfo
 ENUM         | __TypeKind          | t          |         |                 | __TypeKind
 OBJECT       | __Schema            | t          |         |                 | __Schema
 OBJECT       | __Type              | t          |         |                 | __Type
 OBJECT       | __Field             | t          |         |                 | __Field
 OBJECT       | __InputValue        | t          |         |                 | __InputValue
 OBJECT       | __EnumValue         | t          |         |                 | __EnumValue
 ENUM         | __DirectiveLocation | t          |         |                 | __DirectiveLocation
 OBJECT       | __Directive         | t          |         |                 | __Directive
 ENUM         | OrderByDirection    | t          |         |                 | OrderByDirection
 INPUT_OBJECT | FilterType          | f          |         |               2 | IntFilter
 INPUT_OBJECT | FilterType          | f          |         |               3 | FloatFilter
 INPUT_OBJECT | FilterType          | f          |         |               4 | StringFilter
 INPUT_OBJECT | FilterType          | f          |         |               5 | BooleanFilter
 INPUT_OBJECT | FilterType          | f          |         |               6 | DateTimeFilter
 INPUT_OBJECT | FilterType          | f          |         |               7 | BigIntFilter
 INPUT_OBJECT | FilterType          | f          |         |               8 | UUIDFilter
 INPUT_OBJECT | FilterType          | f          |         |               9 | JSONFilter
 OBJECT       | Node                | f          | account |                 | Account
 OBJECT       | Edge                | f          | account |                 | AccountEdge
 OBJECT       | Connection          | f          | account |                 | AccountConnection
 INPUT_OBJECT | OrderBy             | f          | account |                 | AccountOrderBy
 INPUT_OBJECT | FilterEntity        | f          | account |                 | AccountFilter
 INPUT_OBJECT | CreateNode          | f          | account |                 | AccountCreateInput
 INPUT_OBJECT | UpdateNode          | f          | account |                 | AccountUpdateInput
 OBJECT       | UpdateNodeResponse  | f          | account |                 | AccountUpdateResponse
 OBJECT       | DeleteNodeResponse  | f          | account |                 | AccountDeleteResponse
 ENUM         | Enum                | f          |         |                 | PostStatus
(40 rows)

rollback;
